hedera-node/src/main/java/com/hedera/services/records/TxnAwareRecordsHistorian.java

```diff
@@ -300,4 +309,14 @@ private void save(
 				effPayer, baseRecord, consensusSecond, submittingMember);
 		recordCache.setPostConsensus(txnId, baseRecord.getEnumStatus(), expiringRecord);
 	}
+
+
+	private void timestampSidecars(
+			final List<TransactionSidecarRecord.Builder> sidecars,
+			final Instant txnTimestamp
+	) {
+		for (final var sidecar : sidecars) {
+			sidecar.setConsensusTimestamp(MiscUtils.asTimestamp(txnTimestamp));
```

Comment:
_Minor optimization:_ This creates a separate `Timestamp` for every sidecar; but since `Timestamp`s are immutable, there's no harm in just:
```
final var commonTimestamp = MiscUtils.asTimestamp(txnTimestamp);
for (final var sidecar : sidecars) {
    sidecar.setConsensusTimestamp(commonTimestamp);
}

Reply:
Overlooked this one... updated